cmd/cache_serve.go

```diff
@@ -85,33 +82,17 @@ func serveCache(cmd *cobra.Command, _ []string) error {
 	return nil
 }
 
-func serveCacheInternal(ctx context.Context) error {
+func serveCacheInternal(cmdCtx context.Context) error {
 	// Use this context for any goroutines that needs to react to server shutdown
-	ctx, shutdownCancel := context.WithCancel(ctx)
-
-	err := config.InitServer(ctx, config.CacheType)
+	err := config.InitServer(cmdCtx, config.CacheType)
 	cobra.CheckErr(err)
 
-	egrp, ok := ctx.Value(config.EgrpKey).(*errgroup.Group)
+	egrp, ok := cmdCtx.Value(config.EgrpKey).(*errgroup.Group)
 	if !ok {
 		egrp = &errgroup.Group{}
 	}
 
-	egrp.Go(func() error {
```

Comment:
I don't like removing the signal handler from here.  It means that the only way signals are propagated to shutdown Pelican are when xrootd has already been started, pelican forwards the signal to xrootd, and xrootd decides to shutdown.  That's a fairly brittle combination.

Reply:
I thought we had this logic in `cmd/root.go` which is global to all commands we run. Isn't here a duplicated code?

Reply:
No you are right. I got myself confused by the one in the `launcher.go` but cache wasn't launched by that route.